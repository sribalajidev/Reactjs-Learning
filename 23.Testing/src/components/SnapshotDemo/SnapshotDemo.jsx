import React from "react";

export default function SnapshotDemo({ title = "Snapshot Title" }) {
  return (
    <div data-testid="snapshot-demo">
      <h1>{title}</h1>
      <p>This is a snapshot test component</p>
    </div>
  );
}
