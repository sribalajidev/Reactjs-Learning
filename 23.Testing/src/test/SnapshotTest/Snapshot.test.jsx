import React from "react";
import { render } from "@testing-library/react";
import SnapshotDemo from "../../components/SnapshotDemo/SnapshotDemo";

describe("SnapshotDemo Component", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<SnapshotDemo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
