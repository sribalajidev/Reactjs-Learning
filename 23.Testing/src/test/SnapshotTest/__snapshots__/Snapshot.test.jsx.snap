// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SnapshotDemo Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="snapshot-demo"
  >
    <h1>
      Snapshot Title
    </h1>
    <p>
      This is a snapshot test component
    </p>
  </div>
</DocumentFragment>
`;
